R/functions_data.R

```diff
@@ -586,70 +615,70 @@ linkEvents.default <- function(x, ...)
-  }
-  times <- as.POSIXct(times)
-
-  mat <- vector()
-  for (t in times) {
-    net <- as.matrix(network, time = t)
-    raw <- as.data.frame(nodes, time = t)[, attribute]
-    val <- apply(net, 2, function(x) FUN(raw[which(x == 1)], na.rm = TRUE))
-    val[is.nan(val)] <- NA
-    val[val == -Inf] <- NA
-    mat <- rbind(mat, val)
-  }
-
-  row.names(mat) <- as.character(times)
-
-  out <- vector()
-  for (o in as.character(times)[2:length(times)]) {
-    if (!all(mapply(identical, mat[match(o, row.names(mat)), ], mat[match(o, row.names(mat)) - 1, ]))) {
-      out <- rbind(
-        out,
-        data.frame(
-          time = o,
-          node = nodes2$label[which(!mapply(
-            identical,
-            mat[match(o, row.names(mat)), ],
-            mat[match(o, row.names(mat)) - 1, ]
-          ))],
-          replace = mat[o, which(!mapply(
-            identical,
-            mat[match(o, row.names(mat)), ],
-            mat[match(o, row.names(mat)) - 1, ]
-          ))]
-        )
-      )
-    }
-  }
-  row.names(out) <- NULL
-  out$time <- as.POSIXct(as.character(out$time))
-  out$node <- as.character(out$node)
-  return(out)
-}
+# createDist2events <- function(network, nodes, nodes2, attribute, FUN) {
+#   times <- vector("character")
+#   if (attribute %in% attr(nodes, "dynamicAttributes")) {
+#     att.events <- attr(nodes, "events")[which(attr(nodes, "dynamicAttributes") == attribute)]
+#     att.events <- get(att.events)
+#     times <- c(times, as.character(unique(att.events$time)))
+#   }
+#   if (length(attr(network, "events")) > 0) {
+#     net.events <- attr(network, "events")
+#     net.events <- get(net.events)
+#     times <- c(times, as.character(unique(net.events$time)))
+#   }
+#   times <- as.POSIXct(times)
+# 
+#   mat <- vector()
+#   for (t in times) {
+#     net <- as.matrix(network, time = t)
+#     raw <- as.data.frame(nodes, time = t)[, attribute]
+#     val <- apply(net, 2, function(x) FUN(raw[which(x == 1)], na.rm = TRUE))
+#     val[is.nan(val)] <- NA
+#     val[val == -Inf] <- NA
+#     mat <- rbind(mat, val)
+#   }
+# 
+#   row.names(mat) <- as.character(times)
+# 
+#   out <- vector()
+#   for (o in as.character(times)[2:length(times)]) {
+#     if (!all(mapply(identical, mat[match(o, row.names(mat)), ], mat[match(o, row.names(mat)) - 1, ]))) {
+#       out <- rbind(
+#         out,
+#         data.frame(
+#           time = o,
+#           node = nodes2$label[which(!mapply(
+#             identical,
+#             mat[match(o, row.names(mat)), ],
+#             mat[match(o, row.names(mat)) - 1, ]
+#           ))],
+#           replace = mat[o, which(!mapply(
+#             identical,
+#             mat[match(o, row.names(mat)), ],
+#             mat[match(o, row.names(mat)) - 1, ]
+#           ))]
+#         )
+#       )
+#     }
+#   }
+#   row.names(out) <- NULL
+#   out$time <- as.POSIXct(as.character(out$time))
+#   out$node <- as.character(out$node)
+#   return(out)
+# }
```

Comment:
Not being used anymore?

Reply:
I'm wondering about these two functions: `createDist2events` and `createStart`. I'm sure it's not used during preprocessing or estimation. Also, they give warnings regarding not visible binding for a global variable.  But I'm not sure if someone still uses them. I'll delete them and update the news indicating they are no available anymore.

Reply:
Functions deleted in the last commit.